dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientBlockingChannel.java

```diff
@@ -275,10 +279,19 @@ private static final class HttpClientResponse implements Response {
 
         @Override
         public InputStream body() {
+            InputStream snapshot = this.responseBody;
+            if (snapshot == null) {
+                snapshot = createResponseBody();
```

Comment:
entity.getContent creates a new state tracking wrapper around the delegate InputStream for each invocation, by holding a reference to the single delegate, we can be more confident that state is consistent.